python/common/python/time.java

```diff
@@ -89,10 +89,42 @@ public time() {
     }
 
     @org.python.Method(
-            __doc__ = ""
+            __doc__ = "ctime(seconds) -> string\n\n" +
+                      "Convert a time in seconds since the Epoch to a string in local time.\n" +
+                      "This is equivalent to asctime(localtime(seconds)). When the time tuple is\n" +
+                      "not present, current time as returned by localtime() is used.",
+            default_args = {"seconds"}
     )
-    public static org.python.Object ctime() {
-        throw new org.python.exceptions.NotImplementedError("time.ctime() has not been implemented.");
+    public static org.python.Object ctime(org.python.Object seconds) {
+        java.util.Date date;
+        if (seconds == null) {
+            long currentTimeInMillis = System.currentTimeMillis();
+            date = new java.util.Date(currentTimeInMillis);
+        } else {
+            //Type checking the parameter
+            String parameterType = seconds.getClass().getSimpleName();
+            if (!parameterType.equals("Int") && !parameterType.equals("Float")) {
+                if (parameterType.equals("Str")) {
+                    parameterType = "str";
+                } else if (parameterType.equals("List")) {
+                    parameterType = "list";
+                } else if (parameterType.equals("Tuple")) {
+                    parameterType = "tuple";
+                }
+                throw new org.python.exceptions.TypeError("an integer is required (got type " + parameterType + ")");
```

Comment:
As you can see, this will not work for all existing types.
There are several examples in the codebase on how to do this, look at the implementation of `List.insert` for example.